src/printer.js

```diff
@@ -3831,6 +3845,7 @@ function nodeStr(node, options) {
   const alternate = preferred === single ? double : single;
 
   let shouldUseAlternateQuote = false;
+  let canChangeDirectiveQuotes = !rawContent.includes('\\');
```

Comment:
I still don't understand why you need to check if there's a `\`. If you don't change quotes, then it shouldn't matter. If you do change quotes and there are no existing quotes, then having an escaped `\` is fine.